allennlp/tests/data/token_indexers/pretrained_transformer_indexer_test.py

```diff
@@ -3,24 +3,74 @@
 from allennlp.common.testing import AllenNlpTestCase
 from allennlp.data import Token, Vocabulary
 from allennlp.data.token_indexers import PretrainedTransformerIndexer
+from allennlp.data.tokenizers import PretrainedTransformerTokenizer
 
 
 class TestPretrainedTransformerIndexer(AllenNlpTestCase):
-    def test_as_array_produces_token_sequence(self):
+    def test_as_array_produces_token_sequence_bert(self):
         tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
+        allennlp_tokenier = PretrainedTransformerTokenizer("bert-base-uncased")
+
+        # uncased
         indexer = PretrainedTransformerIndexer(model_name="bert-base-uncased")
-        tokens = tokenizer.tokenize("AllenNLP is great")
+        string_specials = "[CLS] AllenNLP is great [SEP]"
+        string_no_specials = "AllenNLP is great"
+        tokens = tokenizer.tokenize(string_specials)
         expected_ids = tokenizer.convert_tokens_to_ids(tokens)
-        allennlp_tokens = [Token(token) for token in tokens]
+        # tokens tokenized with our pretrained tokenizer have indices in them
+        allennlp_tokens = allennlp_tokenier.tokenize(string_no_specials)
         vocab = Vocabulary()
         indexed = indexer.tokens_to_indices(allennlp_tokens, vocab, "key")
         assert indexed["key"] == expected_ids
 
-        tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
+        # cased
         indexer = PretrainedTransformerIndexer(model_name="bert-base-cased")
-        tokens = tokenizer.tokenize("AllenNLP is great")
+        string_specials = "[CLS] AllenNLP is great [SEP]"
+        string_no_specials = "AllenNLP is great"
+        tokens = tokenizer.tokenize(string_specials)
+        expected_ids = tokenizer.convert_tokens_to_ids(tokens)
+        # tokens tokenized with our pretrained tokenizer have indices in them
+        allennlp_tokens = allennlp_tokenier.tokenize(string_no_specials)
+        vocab = Vocabulary()
```

Comment:
I think we might want some expectations on the vocabulary produced. (In a separate test case.) Nothing too heavyweight. Perhaps:

1. Sanity check that just verifies that a short simple sentence produces an expected vocabulary.
2. If it's possible to have tokens that aren't indexed by the tokenizer, introduce some. Verify that the indices of the other tokens don't change.

Reply:
Thanks @brendan-ai2 ! Your first sanity check made me discover https://github.com/allenai/allennlp/issues/3456